File: app/utils/ui/uigo/label.lua
local ____exports = {}
function ____exports.UILabel(url)
    return {
        url = url,
        set_text = function(text)
            label.set_text(url, text)
        end,
        get_text = function()
            return label.get_text(url)
        end,
        set_visible = function(visible)
            local to = go.get(url, "color")
            to.w = visible and 1 or 0
            go.set(url, "color", to)
        end,
        set_scale = function(scale)
            go.set(
                url,
                "scale",
                vmath.vector3(scale, scale, 1)
            )
        end
    }
end
return ____exports

File: src/utils/ui/uigo/label.ts
import { go_ui_lable } from "./interfaces/go_ui";

export function UILabel(url: url) : go_ui_lable {
       
    return {
        url,
        set_text : ( text: string ) => {
            label.set_text(url, text);
        },
        get_text : () : string => {
            return label.get_text(url);
        },
        set_visible : (visible) => {
            const to = go.get(url, 'color') as vmath.vector4;
            
            to.w = visible ? 1 : 0;
            go.set(url, 'color', to);
        }, 
        set_scale: (scale: number) => {
            go.set(url, 'scale', vmath.vector3(scale,scale,1));
        }
    };
}